packages/amplify-graphql-schema-generator/src/datasource-adapter/mysql-datasource-adapter.ts

```diff
@@ -0,0 +1,146 @@
+import { Field, Index } from "../schema-representation";
+import { DataSourceAdapter } from "./datasource-adapter";
+import { knex } from 'knex';
+import { printer } from 'amplify-prompts';
+
+interface MySQLDataSourceConfig {
+  readonly host: string;
+  readonly port: number;
+  readonly database: string;
+  readonly username: string;
+  readonly password: string;
+}
+
+interface MySQLIndex {
+  tableName: string;
+  nonUnique: number;
+  indexName: string;
+  sequence: number;
+  columnName: string;
+  nullable: boolean;  
+}
+
+export class MySQLDataSourceAdapter extends DataSourceAdapter {
+  private dbBuilder: any;
+  private indexes: MySQLIndex[] = [];
+  private readonly PRIMARY_KEY_INDEX_NAME = 'PRIMARY';
+
+  constructor(private config: MySQLDataSourceConfig) {
+    super();
+    this.establishDBConnection();
+    this.loadAllIndexes();
+  }
+
+  private establishDBConnection() {
+    const databaseConfig = {
+      host: this.config.host,
+      database: this.config.database,
+      port: this.config.port,
+      user: this.config.username,
+      password: this.config.password,
+      ssl: { rejectUnauthorized: false},
+    };
+    try {
+      this.dbBuilder = knex({
+        client: 'mysql2',
+        connection: databaseConfig,
+        pool: {
+          min: 5, 
+          max: 30,
+          createTimeoutMillis: 30000,
+          acquireTimeoutMillis: 30000,
+          idleTimeoutMillis: 30000,
+          reapIntervalMillis: 1000,
+          createRetryIntervalMillis: 100
+        },
+        debug: false,
+      });
+    }
+    catch(err) {
+      printer.info(err);
+      throw err;
+    }
+  }
+
+  public async getTablesList(): Promise<string[]> {
+    let result = (await this.dbBuilder.raw("SHOW TABLES"))[0];
```

Comment:
Instead of writing these queries manually can you use https://github.com/knex/knex-schema-inspector?

Reply:
See my comment on the indexes, same thing here.